### PR TITLE
Automatic commit: Move 'advanced-sql-migration-tool-ora-hana' from QA…

### PR DESCRIPTION
**Pull request by javiermartincano**
… to Production

### CODE REVIEW

File: tutorials/advanced-sql-migration-tool-ora-hana/advanced-sql-migration-tool-ora-hana.md [hunk 1]
```diff
@@ -0,0 +1,445 @@
...
+auto_validation: true
+time: 40
+tags: [ tutorial>beginner, programming-tool>sql, products>sap-Hana]
+primary_tag: programming-tool>sql
+---
+
+## Prerequisites
+ - Oracle DB instance (version >= 11), where a new user and SQL objects will be created with a script provided in the tutorial
+ - Oracle JDBC driver (ojdbc7.jar)
+ - SAP HANA or SAP HANA Cloud instance. Minimum versions are 2.0 rev31 and QRC3-2021 respectively.
```
Comment (MichaelCzcz) [inline tutorials/advanced-sql-migration-tool-ora-hana/advanced-sql-migration-tool-ora-hana.md:13]: Please remove period.

File: tutorials/advanced-sql-migration-tool-ora-hana/advanced-sql-migration-tool-ora-hana.md [hunk 2]
```diff
@@ -0,0 +1,445 @@
...
+ - SAP Advanced SQL Migration tool installed and configured properly as indicated in the installation guide:  <https://dam.sap.com/mac/app/p/pdf/asset/preview/cLmo5vJ?ltr=a>
+ - Java >= 1.8 installed in the host where the migration tool is executed
+
+## Details
+### You will learn
+- Convert Oracle schema to SAP HANA
+
+The SAP Advanced SQL Migration tool is SAP's migration tool to migrate custom applications running on top of non-SAP databases into SAP databases.
+
+![Select Data Source](ad-sql-mig.png)
```
Comment (MichaelCzcz) [inline tutorials/advanced-sql-migration-tool-ora-hana/advanced-sql-migration-tool-ora-hana.md:23]: Add border

File: tutorials/advanced-sql-migration-tool-ora-hana/advanced-sql-migration-tool-ora-hana.md [hunk 3]
```diff
@@ -0,0 +1,445 @@
...
+> Every migration tool command creates a log file located under subfolder called `reports` (the folder is created automatically if it doesn't exit).
+
+
+[DONE]
+[ACCORDION-END]
+
+
+[ACCORDION-BEGIN [Step 3: ](Extract data model and SQL objects)]
+
+Data model and SQL objects extraction can be done either connecting to the source database via the migration tool gateway or using text files with object definitions (DDLS). Section 1 is mandatory and you will go through the extraction via the migration tool gateway. Then section 2 is optional, do it if you want to see the extraction using a text file as well.
```
Comment (MichaelCzcz) [inline tutorials/advanced-sql-migration-tool-ora-hana/advanced-sql-migration-tool-ora-hana.md:130]: "...can be done either by..."

File: tutorials/advanced-sql-migration-tool-ora-hana/advanced-sql-migration-tool-ora-hana.md [hunk 4]
```diff
@@ -0,0 +1,445 @@
...
+
+> **IMPORTANT:** It is recommended for you to try both methods, go through the gateway first in section 1 and then go through optional steps in section 2
+
+### Extract using the migration tool gateway
+
+The migration tool gateway is usually executed in a separate host, but for simplicity in this tutorial it will be executed in the same host where the migration tool is installed, notice that java version 1.8 minimum is required to run the gateway.
```
Comment (MichaelCzcz) [inline tutorials/advanced-sql-migration-tool-ora-hana/advanced-sql-migration-tool-ora-hana.md:136]: Java

File: tutorials/advanced-sql-migration-tool-ora-hana/advanced-sql-migration-tool-ora-hana.md [hunk 5]
```diff
@@ -0,0 +1,445 @@
...
+
+    For example:
+
+    `c://sapdbmtk 32# sapdbmtk extract C:/tmp/Advanced_SQL_Mig_tutorial <ora_intance> XXX.XXX.XXX.XXX 1521 <oracle_user> <system_passwd>`
+
+    NOTE: the Oracle user you specify for this command needs to have read permissions on the catalogs, it will not perform any modification in the source database, SELECT_CATALOG_ROLE is enough for this purpose.
+
+    > **IMPORTANT:** windows firewall or antivirus software can prevent the migration tool connecting to the gateway, if you have connectivity issues try stopping/configuring firewall/antivirus
+
+6. When you execute `extract` option for the first time in a new project folder you will get 2 popup windows to confirm that compliance with 3rd party rights has been verified. When each of those 2 popup windows appears just read and click on **I confirm**, then rerun the command again (you will have to rerun the command twice the first time).
```
Comment (MichaelCzcz) [inline tutorials/advanced-sql-migration-tool-ora-hana/advanced-sql-migration-tool-ora-hana.md:177]: "...the ´extract´ option..."

File: tutorials/advanced-sql-migration-tool-ora-hana/advanced-sql-migration-tool-ora-hana.md [hunk 6]
```diff
@@ -0,0 +1,445 @@
...
+
+    ![Select Data Source](extract-1.png)
+
+    The `extract` command is done when you get the prompt again.
+
+    ![Select Data Source](extract-2.png)
+
+    > The complete output for the extract is located under `<project_folder>\reports` in a file named <ORA_SERVER_name\>.SQL-Extraction.Oracle-HANA.YYYYMMMDD.hhmiss.log
+
+    With `extract` command the migration tools builds its own metadata about the data model and SQL objects found in the source database. A new subfolder called `extracted` is created under the project folder, in it you will find a file for every SQL object like a stored procedure/function/trigger. and another subfolder called `catalog` storing files with information about users, tables, columns and so forth.
```
Comment (MichaelCzcz) [inline tutorials/advanced-sql-migration-tool-ora-hana/advanced-sql-migration-tool-ora-hana.md:191]: "With the ´extract´" command..."

Either replace that period between /trigger and "and" with a comma or start with upper case "A".

File: tutorials/advanced-sql-migration-tool-ora-hana/advanced-sql-migration-tool-ora-hana.md [hunk 7]
```diff
@@ -0,0 +1,445 @@
...
+
+    ![Select Data Source](extracted-folder.png)
+
+### Extract using text file containing object definitions (OPTIONAL)
+
+> Before proceeding with this optional step you have to delete subfolder called `extracted` under the project folder, it was generated when running `extract` option with the Gateway and will generate it again now
+
+1. DDL file/s preparation.
+
+    The script provided and used to create the Oracle user and data model in step 1 can be used for this purpose as well.
```
Comment (MichaelCzcz) [inline tutorials/advanced-sql-migration-tool-ora-hana/advanced-sql-migration-tool-ora-hana.md:201]: "The script that is provided..."

File: tutorials/advanced-sql-migration-tool-ora-hana/advanced-sql-migration-tool-ora-hana.md [hunk 8]
```diff
@@ -0,0 +1,445 @@
...
+
+    > Subfolder `extracted` is created under the project folder in the same way it was created for the `extract` command in section 1.
+
+
+[VALIDATE_2]
+[ACCORDION-END]
+
+[ACCORDION-BEGIN [Step 4: ](Convert to SAP HANA)]
+
+Once the source data model and SQL object are extracted, the next step is to run the conversion to the target database -- SAP HANA in this case. For that, you have to use migration tool `convert` command with the following parameters:
```
Comment (MichaelCzcz) [inline tutorials/advanced-sql-migration-tool-ora-hana/advanced-sql-migration-tool-ora-hana.md:231]: "...you have to use the migration tool..."

File: tutorials/advanced-sql-migration-tool-ora-hana/advanced-sql-migration-tool-ora-hana.md [hunk 9]
```diff
@@ -0,0 +1,445 @@
...
+
+For example:
+
+`c://sapdbmtk 32# sapdbmtk convert C:/tmp/Advanced_SQL_Mig_tutorial`
+
+When you run the `convert` command, you can see this output in the command line window.
+
+![Select Data Source](convert-1.png)
+
+And once is it completed, you will get the prompt with this output:
```
Comment (MichaelCzcz) [inline tutorials/advanced-sql-migration-tool-ora-hana/advanced-sql-migration-tool-ora-hana.md:243]: "And once it is completed..."

File: tutorials/advanced-sql-migration-tool-ora-hana/advanced-sql-migration-tool-ora-hana.md [hunk 10]
```diff
@@ -0,0 +1,445 @@
...
+
+![Select Data Source](convert-2.png)
+
+> The complete output for the extract is located under `<project_folder>\reports` in a file named `SQL-Conversion.Oracle-HANA.YYYYMMMDD.hhmiss.log`.
+
+When `convert` is executed, 2 new subfolders are created in the project folder: `converted_orig` and `scripts`.
+
+- In `converted_orig` you can find a `.sqlscript` file with the converted code for each procedure/function/trigger and a `.convertlog` file per object as well with conversion details.
+
+- In `scripts` you can find all the ready to run shell scripts and SQL scripts generated by the migration tool to be directly executed against the target database. Notice that a copy of `converted_orig` folder is created under `scripts` with name `converted`.
```
Comment (MichaelCzcz) [inline tutorials/advanced-sql-migration-tool-ora-hana/advanced-sql-migration-tool-ora-hana.md:253]: ready-to-run

File: tutorials/advanced-sql-migration-tool-ora-hana/advanced-sql-migration-tool-ora-hana.md [hunk 11]
```diff
@@ -0,0 +1,445 @@
...
+
+![Select Data Source](scripts-folder.png)
+
+[VALIDATE_3]
+[ACCORDION-END]
+
+[ACCORDION-BEGIN [Step 5: ](Run converted scripts)]
+
+After running `convert` command two shell scripts called `run_ddl_scripts.sh` and `run_sql_scripts.sh.` have been created under `<project_folder>\scripts`, they are ready to be executed against SAP HANA. The former will create the data model (tables/views/indexes/synonyms …etc.) and the latter will create SQL objects like stored procedures, functions, views,...etc.
```
Comment (MichaelCzcz) [inline tutorials/advanced-sql-migration-tool-ora-hana/advanced-sql-migration-tool-ora-hana.md:262]: "After running the..."

File: tutorials/advanced-sql-migration-tool-ora-hana/advanced-sql-migration-tool-ora-hana.md [hunk 12]
```diff
@@ -0,0 +1,445 @@
...
+    ![Select Data Source](hana-studio.png)
+
+
+
+[VALIDATE_4]
+[ACCORDION-END]
+
+[ACCORDION-BEGIN [Step 6: ](Check Oracle and SAP HANA behavior)]
+
+Run some SQL stuff in Oracle to force a couple of errors and see how Oracle behaves, then run exactly the same in SAP HANA to confirm it is working similarly and getting the same behavior after the conversion.
```
Comment (MichaelCzcz) [inline tutorials/advanced-sql-migration-tool-ora-hana/advanced-sql-migration-tool-ora-hana.md:299]: "...and showing the same..."

File: tutorials/advanced-sql-migration-tool-ora-hana/advanced-sql-migration-tool-ora-hana.md [hunk 13]
```diff
@@ -0,0 +1,445 @@
...
+
+1. Oracle
+
+    Open the provided text file with the Oracle data model (`APPDEMO2.sql`) and check the SQL code for a stored procedure called `appdemo2.insert_author`.
+
+    Notice it is checking different things before performing the real insert into `appdemo2.authors` table and raising errors when apply -- for example, it checks that `state` value is no longer than 2 characters and `phone` has an specific format with no letters (it is being validated using an Oracle function).
```
Comment (MichaelCzcz) [inline tutorials/advanced-sql-migration-tool-ora-hana/advanced-sql-migration-tool-ora-hana.md:305]: "Notice that..."

File: tutorials/advanced-sql-migration-tool-ora-hana/advanced-sql-migration-tool-ora-hana.md [hunk 14]
```diff
@@ -0,0 +1,445 @@
...
+    'Madrid',
+    'MAD',
+    'Spain',
+    '28043',
+    null
+    )
+    /
+    ```
+
+    It fails with following error:
```
Comment (MichaelCzcz) [inline tutorials/advanced-sql-migration-tool-ora-hana/advanced-sql-migration-tool-ora-hana.md:326]: "..with the following error"

File: tutorials/advanced-sql-migration-tool-ora-hana/advanced-sql-migration-tool-ora-hana.md [hunk 15]
```diff
@@ -0,0 +1,445 @@
...
+    null
+    )
+    /
+    ```
+
+    Run the new stored procedure call, and a second error appears:
+
+    ![Select Data Source](ora-sql-2.png)
+
+    Modify the stored procedure call again removing character `a` from `phone` field:
```
Comment (MichaelCzcz) [inline tutorials/advanced-sql-migration-tool-ora-hana/advanced-sql-migration-tool-ora-hana.md:351]: "...by removing..."

File: tutorials/advanced-sql-migration-tool-ora-hana/advanced-sql-migration-tool-ora-hana.md [hunk 16]
```diff
@@ -0,0 +1,445 @@
...
+    null
+    )
+    /
+    ```
+
+    It fails with following error:
+
+    ![Select Data Source](ora-sql-1.png)
+
+    Modify the stored procedure call removing last letter (`D`) from `state` field:
```
Comment (MichaelCzcz) [inline tutorials/advanced-sql-migration-tool-ora-hana/advanced-sql-migration-tool-ora-hana.md:330]: "...by removing..."

File: tutorials/advanced-sql-migration-tool-ora-hana/advanced-sql-migration-tool-ora-hana.md [hunk 17]
```diff
@@ -0,0 +1,445 @@
...
+    'Madrid',
+    'MAD',
+    'Spain',
+    '28043',
+    null
+    )
+    go
+    ```
+
+    Notice that the same error we had in Oracle is appearing in SAP HANA as well.
```
Comment (MichaelCzcz) [inline tutorials/advanced-sql-migration-tool-ora-hana/advanced-sql-migration-tool-ora-hana.md:398]: "...you had..."

File: tutorials/advanced-sql-migration-tool-ora-hana/advanced-sql-migration-tool-ora-hana.md [hunk 18]
```diff
@@ -0,0 +1,445 @@
...
+
+    ![Select Data Source](hana-sql-1.png)
+
+    Modify the stored procedure call removing `D` from `state` field:
```
Comment (MichaelCzcz) [inline tutorials/advanced-sql-migration-tool-ora-hana/advanced-sql-migration-tool-ora-hana.md:402]: "...by removing..."

File: tutorials/advanced-sql-migration-tool-ora-hana/advanced-sql-migration-tool-ora-hana.md [hunk 19]
```diff
@@ -0,0 +1,445 @@
...
+    null
+    )
+    go
+    ```
+
+    Run the stored procedure call again and the second error pointing to an invalid phone appears.
+
+    ![Select Data Source](hana-sql-2.png)
+
+    Modify the stored procedure call removing character `a` from the phone number:
```
Comment (MichaelCzcz) [inline tutorials/advanced-sql-migration-tool-ora-hana/advanced-sql-migration-tool-ora-hana.md:423]: "...by removing..."

**Review by MichaelCzcz — CHANGES_REQUESTED** [review]
I couldn't test validation in the QA system due to some technical problems. So, I tested it locally in Atom. They all seem to work properly.

Please fix the issues that I have commented on. I will merge your PR afterwards as requested by Daniel.

File: tutorials/advanced-sql-migration-tool-ora-hana/advanced-sql-migration-tool-ora-hana.md (hunk 1 above)
Comment (javiermartincano) [reply tutorials/advanced-sql-migration-tool-ora-hana/advanced-sql-migration-tool-ora-hana.md:13]: Done

File: tutorials/advanced-sql-migration-tool-ora-hana/advanced-sql-migration-tool-ora-hana.md (hunk 2 above)
Comment (javiermartincano) [reply tutorials/advanced-sql-migration-tool-ora-hana/advanced-sql-migration-tool-ora-hana.md:23]: Done

File: tutorials/advanced-sql-migration-tool-ora-hana/advanced-sql-migration-tool-ora-hana.md (hunk 3 above)
Comment (javiermartincano) [reply tutorials/advanced-sql-migration-tool-ora-hana/advanced-sql-migration-tool-ora-hana.md:130]: Done

File: tutorials/advanced-sql-migration-tool-ora-hana/advanced-sql-migration-tool-ora-hana.md (hunk 4 above)
Comment (javiermartincano) [reply tutorials/advanced-sql-migration-tool-ora-hana/advanced-sql-migration-tool-ora-hana.md:136]: Done

File: tutorials/advanced-sql-migration-tool-ora-hana/advanced-sql-migration-tool-ora-hana.md (hunk 5 above)
Comment (javiermartincano) [reply tutorials/advanced-sql-migration-tool-ora-hana/advanced-sql-migration-tool-ora-hana.md:177]: Done

File: tutorials/advanced-sql-migration-tool-ora-hana/advanced-sql-migration-tool-ora-hana.md (hunk 15 above)
Comment (javiermartincano) [reply tutorials/advanced-sql-migration-tool-ora-hana/advanced-sql-migration-tool-ora-hana.md:351]: Done

File: tutorials/advanced-sql-migration-tool-ora-hana/advanced-sql-migration-tool-ora-hana.md (hunk 16 above)
Comment (javiermartincano) [reply tutorials/advanced-sql-migration-tool-ora-hana/advanced-sql-migration-tool-ora-hana.md:330]: Done

File: tutorials/advanced-sql-migration-tool-ora-hana/advanced-sql-migration-tool-ora-hana.md (hunk 17 above)
Comment (javiermartincano) [reply tutorials/advanced-sql-migration-tool-ora-hana/advanced-sql-migration-tool-ora-hana.md:398]: Done

File: tutorials/advanced-sql-migration-tool-ora-hana/advanced-sql-migration-tool-ora-hana.md (hunk 18 above)
Comment (javiermartincano) [reply tutorials/advanced-sql-migration-tool-ora-hana/advanced-sql-migration-tool-ora-hana.md:402]: Done

File: tutorials/advanced-sql-migration-tool-ora-hana/advanced-sql-migration-tool-ora-hana.md (hunk 19 above)
Comment (javiermartincano) [reply tutorials/advanced-sql-migration-tool-ora-hana/advanced-sql-migration-tool-ora-hana.md:423]: Done

File: tutorials/advanced-sql-migration-tool-ora-hana/advanced-sql-migration-tool-ora-hana.md (hunk 2 above)
Comment (MichaelCzcz) [reply tutorials/advanced-sql-migration-tool-ora-hana/advanced-sql-migration-tool-ora-hana.md:23]: There is still no border.
It should be !Select Data Source

**Review by MichaelCzcz — CHANGES_REQUESTED** [review]
Please address all issues.

File: tutorials/advanced-sql-migration-tool-ora-hana/advanced-sql-migration-tool-ora-hana.md (hunk 2 above)
Comment (javiermartincano) [reply tutorials/advanced-sql-migration-tool-ora-hana/advanced-sql-migration-tool-ora-hana.md]: I see ... what I did was to add the border in the picture itself. I have removed the border from the picture and add that second "!" you are pointing

**Pull request merged**